### PR TITLE
create separate ack manager for transfer queue, for failover, as well as ack manager base

### PR DESCRIPTION
**Pull request by wxing1292**
UT in next PR

### CODE REVIEW

File: service/history/queueProcessor.go [hunk 1]
```diff
@@ -355,7 +355,8 @@ MoveAckLevelLoop:
 	}
 
 	// Always update ackLevel to detect if the shared is stolen
-	if err := a.shard.UpdateTransferAckLevel(updatedAckLevel); err != nil {
+	clusterName := a.shard.GetService().GetClusterMetadata().GetCurrentClusterName()
```
Comment (samarabbas) [inline service/history/queueProcessor.go]: Can we store the clusterName when queueProcessor is created?

File: service/history/queueAckMgrBase.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+		t.outstandingTasks[queueTask.GetTaskID()] = false
+	}
+}
+
+func (t *queueAckMgrBase) completeTask(queueTask queueTaskInfo) {
+	t.outstandingTasks[queueTask.GetTaskID()] = true
+}
+
+func (t *queueAckMgrBase) updateAckLevel(maxAckLevel int64) int64 {
+	t.metricsClient.IncCounter(metrics.TimerQueueProcessorScope, metrics.AckLevelUpdateCounter)
```
Comment (samarabbas) [inline service/history/queueAckMgrBase.go]: TimerQueueProcessorScope?

File: service/history/queueAckMgrBase.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+	initialAckLevel := t.ackLevel
+	updatedAckLevel := t.ackLevel
+
+MoveAckLevelLoop:
```
Comment (samarabbas) [inline service/history/queueAckMgrBase.go]: original ackLevel does this within a lock?

File: service/history/queueAckMgrBase.go [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+	for current := initialAckLevel + 1; current <= maxAckLevel; current++ {
+		acked := t.outstandingTasks[current]
```
Comment (samarabbas) [inline service/history/queueAckMgrBase.go]: ackLevels can have holes.  This will not work if we have holes in the task ids.

Comment (wxing1292) [reply service/history/queueAckMgrBase.go]: good catch

File: service/history/queueAckMgrBase.go (hunk 3 above)
Comment (wxing1292) [reply service/history/queueAckMgrBase.go]: everything which use this base will have its dedicated lock.

File: service/history/queueAckMgrBase.go (hunk 2 above)
Comment (wxing1292) [reply service/history/queueAckMgrBase.go]: ok

**Pull request closed without merging**